File: json/json-annotation-processor/src/test/java/ru/tinkoff/kora/json/annotation/processor/dto/DtoWithNullableFields.java
package ru.tinkoff.kora.json.annotation.processor.dto;

import ru.tinkoff.kora.json.common.annotation.JsonField;
import ru.tinkoff.kora.json.common.annotation.JsonReader;

import javax.annotation.Nullable;

@JsonReader
public record DtoWithNullableFields(@JsonField("field_1") String field1, int field4, @Nullable String field2, @Nullable String field3) {}

File: json/json-symbol-processor/src/test/kotlin/ru/tinkoff/kora/json/ksp/dto/DtoWithNullableFields.kt
package ru.tinkoff.kora.json.ksp.dto

import ru.tinkoff.kora.json.common.annotation.JsonField
import ru.tinkoff.kora.json.common.annotation.JsonReader
import java.util.*

@JsonReader
data class DtoWithNullableFields(
    @JsonField("field_1") val field1: String,
    val field4: Int,
    val field2: String?,
    val field3: String?
)
